File: 1-Semestre/Desarrollo de algoritmos/Actividades/Diagramas de flujo-Python (Semana 5)/Ejercicio 12.py
'''
Calcular el sueldo líquido de un trabajador.
Los elementos a considerar son:
    - sueldo base
    - fondo de pensión (20%)
    - fondo de salud (7%)
Estos tres valores son ingresados por el usuario.
'''

print("Este programa calcula el sueldo líquido de un trabajador.")
base = int(input("Ingrese el sueldo base: "))
afp = int(input("Ingrese el fondo de pensión (poner 0 para tomar el 20%): "))
salud = int(input("Ingrese el fondo de salud (poner 0 para tomar el 7%): "))

while base <= 0 or afp < 0 or salud < 0:
    print("[!] Debe ingresar valores positivos.")
    base = int(input("Ingrese el sueldo base: "))
    afp = int(input("Ingrese el fondo de pensión (poner 0 para tomar el 20%): "))
    salud = int(input("Ingrese el fondo de salud (poner  en blanco para tomar el 7%): "))

if afp == 0:
    afp = base * 0.20

if salud == 0:
    salud = base * 0.07

liquido = base - (afp + salud)
print(f"El sueldo líquido es de ${liquido}")

File: 2-Semestre/Taller de Programación/Personal/Python-C#/Ejercicio 12.cs
/*
Calcular el sueldo líquido de un trabajador.
Los elementos a considerar son:
    - sueldo base
    - fondo de pensión (20%)
    - fondo de salud (7%)
Estos tres valores son ingresados por el usuario.
*/

using System;

namespace Ejercicio_12
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Este programa calcula el sueldo líquido de un trabajador\n");

            double Base, afp, salud;

            do
            {
                Console.Write("Ingrese el sueldo base: ");
                Base = Convert.ToDouble(Console.ReadLine());

                if (Base <= 0)
                {
                    Console.WriteLine("[!] El suelo base no puede ser menor o igual a 0");
                }
            } while (Base <= 0);

            do
            {
                Console.Write("Ingrese el fondo de pensión (poner 0 para tomar el 20% del sueldo base): ");
                afp = Convert.ToDouble(Console.ReadLine());

                if (afp < 0)
                {
                    Console.WriteLine("[!] El fondo de pensión no puede ser menor a 0");
                }
            } while (afp < 0);

            do
            {
                Console.Write("Ingrese el fondo de salud (poner 0 para tomar el 7% del sueldo base): ");
                salud = Convert.ToDouble(Console.ReadLine());

                if (salud < 0)
                {
                    Console.WriteLine("[!] El fondo de salud no puede ser menor a 0");
                }
            } while (salud < 0);

            if (afp == 0)
            {
                afp = Base * 0.20;
            }

            if (salud == 0)
            {
                salud = Base * 0.07;
            }

            double liquido = Base - (afp + salud);
            Console.WriteLine($"El sueldo líquido es de ${liquido}");
        }
    }
}